### PR TITLE
Fix the metadata mismatch

### PR DESCRIPTION
**Pull request by pncnmnp**
Based on Issue #17

See the analysis at: https://github.com/deepjyoti30/ytmdl/issues/272
to understand why this approach is more accurate than the native ytmdl implementation.

### CODE REVIEW

File: radio.py [hunk 1]
```diff
@@ -394,29 +398,59 @@ def on_this_day(self):
...
+        return 0
+
+    def postprocess_music(self, song, is_local):
+        """
+        Generate the wav file for music and sandwich it between the intro and outro
+        """
+        song_path = song if is_local else f"{self.audio_dir}/song.mp3"
+        sound = AudioSegment.from_mp3(song_path)
+
+        # Rename the previous wav file (outro) to current index
```
Comment (pncnmnp) [inline radio.py:441]: Rename the previous outro file to current index

File: radio.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 from pathlib import Path
 import re
 import subprocess
+import time
```
Comment (pncnmnp) [inline radio.py:20]: Update `requirements.txt`

File: radio.py [hunk 3]
```diff
@@ -25,8 +26,10 @@
 from eyed3.id3.frames import ImageFrame
 from feedparser import parse
 from ffmpy import FFmpeg
+import itunespy
```
Comment (pncnmnp) [inline radio.py:29]: Update `requirements.txt`

File: radio.py [hunk 4]
```diff
@@ -553,13 +587,26 @@ def music_meta(self, song, artist, is_local, start=True):
             artist, song = metadata.tag.artist, metadata.tag.title
             genre = "The next song is from your personal collection. "
         else:
+            # Search for the song metadata in the fetched mp3 file
```
Comment (pncnmnp) [inline radio.py:590]: Redundant

File: radio.py [hunk 5]
```diff
@@ -553,13 +587,26 @@ def music_meta(self, song, artist, is_local, start=True):
...
+            fetched_artist = eyed3.load(f"{self.audio_dir}/song.mp3").tag.artist
+
+            # Compare the artist name fetched from the metadata
```
Comment (pncnmnp) [inline radio.py:593]: Redundant

File: radio.py [hunk 6]
```diff
@@ -553,13 +587,26 @@ def music_meta(self, song, artist, is_local, start=True):
...
-            except TypeError:
-                return None
-            artist = artist if artist else info["artistName"]
-            song = song if song else info["trackName"]
-            genre = f'The next song is from the world of {info["primaryGenreName"]}. '
+                itunes_metadata = itunespy.search_track(song, country="US", limit=100)
+            except:
+                time.sleep(80)
+                itunes_metadata = itunespy.search_track(song, country="US", limit=100)
+            most_accurate = sorted(
```
Comment (pncnmnp) [inline radio.py:601]: Seems better.

```python
            most_accurate = sorted(
                [song_info.json for song_info in itunes_metadata],
                key=lambda song_info: nltk.edit_distance(
                    song_info["artistName"], fetched_artist
                ),
            )[0]
```

**Comment by ology** [comment]
I didn't test this in different ways - just the latest config and data bits I had. Works perfectly. 👍🏻

**Comment by pncnmnp** [comment]
@ology, thank you so much, Gene! I will write some relevant unit tests this week, and also tidy it up a bit. I'm optimistic that we can merge this PR by Friday.

**Comment by pncnmnp** [comment]
LGTM! Tests seem to work. Merging this PR.
If we find any future issues, we can open a new issue in the future.

**Pull request merged**